hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/BucketIdentifier.java

```diff
@@ -61,13 +60,11 @@ public static List<String> getHashKeys(HoodieKey hoodieKey, String indexKeyField
   }
 
   protected static List<String> getHashKeys(String recordKey, String indexKeyFields) {
-    return !recordKey.contains(":") ? Collections.singletonList(recordKey) :
```

Comment:
Checks like this are moved in one place, `KeyGenUtil::extractRecordKeysByFields`, and fixed by check both `:` and `,` to decide is it a complex key.